libqtile/notify.py

```diff
@@ -26,6 +26,10 @@
     If dbus is available, this module implements a
     org.freedesktop.Notifications service.
 """
+from collections import OrderedDict
```

Comment:
As of python 3.7 dict is ordered by order of insertion. It looks like this code relies on insertion order in the same way so OrderedDict might not be required. Python 3.7 is the minimum python version supported by qtile.

The only thing you would need to change about the code is your use of reversed. values will need to be converted to a list before it can be reversed. In python 3.8 that isn't the case, so a comment for removal later might be nice =)

Reply:
Oh here is something I didn't know about dict in newer versions.

I will drop the OrderedDict.

What I did not get is the comment about reversing. I am using reverse only to get the last key (which is a notification id). I don't actually care about the value (now that I see it line [217](https://github.com/qtile/qtile/blob/4127cf35bcce2a989647cfade159c1571edf6c51/libqtile/notify.py#L217) is obsolette.

Also did not get the part about python 3.8 (maybe because I did not get the comment about the values in reverse).

Thanks for the comment.

Reply:
I also did not know this 👀